### ISSUE TITLE
evaluate problem

### ISSUE THREAD

**Issue by lingheng123** [PRBonn/4DMOS#20]
I wanted to evaluate but ran into the following situation. The visualization works normally.
![IMG_4687](https://user-images.githubusercontent.com/88220400/229708311-e7c19f21-0cd8-4495-9024-d63e6eade5da.JPG)

**Comment by benemer** [comment]
It could not find any predictions. What's the content of `/home/robot/4DMOS/predictions/5_scans/poses/labels/strategy_non_overlapping_0.1`?

**Comment by lingheng123** [comment]
![IMG_4688](https://user-images.githubusercontent.com/88220400/229712135-c79adafb-e399-45d4-879c-b8353a00803b.JPG)
![IMG_4689](https://user-images.githubusercontent.com/88220400/229712154-029562ba-2c67-46ff-8f86-cbd100d7384c.JPG)

**Comment by lingheng123** [comment]
when i use  -s test,i could not find any label
![63E6BE237BE68A5F3D4F711F8F221DB1](https://user-images.githubusercontent.com/88220400/229712931-4818a088-976c-4188-ad03-83bec0c1adb1.png)

**Comment by benemer** [comment]
What are the sequences you have in your dataset? And how is your split in `config/semantic-kitti-mos.yaml`?

The default one is

```yaml
split: # sequence numbers
  train:
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
    - 9
    - 10
  valid:
    - 8
  test:
    - 11
    - 12
    - 13
    - 14
    - 15
    - 16
    - 17
    - 18
    - 19
    - 20
    - 21
    ```

so these sequences need to be in both the dataset and your predictions when evaluating.

**Comment by lingheng123** [comment]
The data set is set according to the format of the sematic-kitti data set, and the configuration file is unchanged.My prediction sequence is 11 to 21

**Comment by benemer** [comment]
Okay. Well in that case, this does not look to be related to 4DMOS. Please check again that you have the predictions and labels specified in the config for the split you select.

For example, in the screenshot of your `sequences` folder, there is not folder for the validation set 08. That's why it can not find the predictions for the validation set.

**Comment by lingheng123** [comment]
Yes,when I execute python scripts/confidences_to_labels.py -p predictions/ID/POSES,I choose default sequences (11 to21).That’s why there is no 8 sequence.Can I only use 8 sequence when evaluating IoU?

**Comment by benemer** [comment]
Note that you can not evaluate MOS for the test set, because the labels are not public. That's why it can not find them when you run with `-s test`. If you want to get a result here you need to submit your predictions to our benchmark in codalab (see the `semantic-kitti-api` repo for details).

If you want to evaluate the IoU of the validation sequence 08, you can specify it in the `predict_confidences.py` by passing `-seq 8` or change the test split in the config to `08`. Make sure to also select the sequence in the `confidences_to_labels.py` script using the `-seq` flag.

**Comment by lingheng123** [comment]
Excuse me, why can't I use pose.txt for 8 sequence to predict?

**Comment by benemer** [comment]
Where did I say you can't use the `poses.txt`? Sorry I don't get what you mean

**Comment by lingheng123** [comment]
Sorry, did I mean to say that I should use suma pose-txt without loop closure?

**Comment by benemer** [comment]
No, this issue is about you having troubles to evaluate the IoU on the predictions. What are you referring to?

In another issue I said that the SuMa poses with loop closing cause false positives on sequence 08 and that you should use the poses without loop closing. But this is another issue and is not related to this one. Please check the other issues and open a new one if you can not find the answer there.

Back to this issue, is it solved now?

**Comment by lingheng123** [comment]
yes，I can evalute 8 sequence to get IoU(0.57).I'm sorry to bother you.

**Comment by benemer** [comment]
Great! No worries. If you now use the SuMa poses without loop closing for sequence 08 you will get the result reported in the paper. I will close this issue now since it is solved. Just open a new one if another problem arises.
